### PR TITLE
Make cross-compiling with bzlmod possible, added example

### PR DESCRIPTION
**Pull request by jondo2010**
Before it wasn't possible to add your desired cross-compilation targets to `extra_target_triples` if they also happened to be in `DEFAULT_TOOLCHAIN_TRIPLES`.

### CODE REVIEW

File: examples/bzlmod/cross_compile/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,33 @@
+load("@rules_rust//rust:defs.bzl", "rust_binary")
+
+package(default_visibility = ["//visibility:public"])
+
+rust_binary(
+    name = "hello_world_aarch64",
```
Comment (illicitonion) [inline examples/bzlmod/cross_compile/BUILD.bazel:6]: Can you throw in a `sh_test` or `rust_test` which takes this as a data dependency, runs `file` on it, and asserts on the output to show that it was _actually_ compiled for the correct target platform?

**Review by illicitonion — COMMENTED** [review]
This looks great, thanks! Just one question about testing

File: examples/bzlmod/cross_compile/BUILD.bazel (hunk 1 above)
Comment (jondo2010) [reply examples/bzlmod/cross_compile/BUILD.bazel:6]: Done!

**Review by illicitonion — APPROVED** [review]
Thanks!

**Comment by jondo2010** [comment]
@illicitonion any chance of landing this?

**Review by illicitonion — APPROVED** [review]

**Pull request merged**